### PR TITLE
Implementing changes to allow compilation under MSVC 2013

### PR DESCRIPTION
**Pull request by imikejackson**

### CODE REVIEW

**Comment by danshapero** [comment]
I've been out of the country for a spell, give me a few days and I'll get to this.

File: src/predicates.c [hunk 1]
```diff
@@ -1105,7 +1105,7 @@ REAL *e;
 /*                                                                           */
 /*****************************************************************************/
 
-REAL orient2dfast(const REAL * pa, const REAL * pb, const REAL * pc)
+predicates_EXPORT REAL orient2dfast(const REAL * pa, const REAL * pb, const REAL * pc)
```
Comment (danshapero) [inline src/predicates.c]: The export declaration is [only necessary in header files](https://stackoverflow.com/questions/557317/is-declspecdllexport-needed-in-cpp-files).

File: src/constants.h [hunk 2]
```diff
@@ -2,6 +2,8 @@
 #ifndef CONSTANTS_H
 #define CONSTANTS_H
 
+#include "predicatesDLLExport.h"
```
Comment (danshapero) [inline src/constants.h]: This include shouldn't be necessary -- we're not exporting anything from this header.

File: include/predicatesDLLExport.h [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+configures to build a shared library. If you are going to use
+another build system on windows or create the visual studio
+projects by hand you need to define predicates_EXPORTS when
+building the predicates DLL on windows.
+*/
+
+
+#if defined (predicates_EXPORTS)  /* Compiling the predicates DLL/Dylib */
+  #if defined (_MSC_VER)  /* MSVC Compiler Case */
+    #define  predicates_EXPORT __declspec(dllexport)
```
Comment (danshapero) [inline include/predicatesDLLExport.h]: Can you change this to `PREDICATES_EXPORT`? I usually stick to all-caps for preprocessor defines.

Comment (imikejackson) [reply include/predicatesDLLExport.h]: yes.

**Comment by imikejackson** [comment]
I have pushed the requested changes.

**Comment by danshapero** [comment]
Looks good, please rebase and I'll merge.

**Comment by imikejackson** [comment]
OK. Think I rebased correctly.

**Pull request merged**